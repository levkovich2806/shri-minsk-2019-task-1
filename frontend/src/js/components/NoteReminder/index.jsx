import React from "react";

import styles from "./index.module.scss";
import { getReadableReminder } from '../../utils/utils';

function NoteReminder(props) {
  return (
    <div className={styles.reminder}>
      <div className={styles.reminder__image}>
        <img src="./images/reminder20.png" alt="Напоминание" />
      </div>
      <div className={styles.reminder__text}>
        {getReadableReminder(props.reminder)}
      </div>
    </div>
  );
}

export default NoteReminder;
