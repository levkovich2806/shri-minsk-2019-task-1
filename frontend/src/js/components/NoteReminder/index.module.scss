.reminder {
  position: relative;
  display: flex;
  align-items: center;
  background-color: #db5341;
  border-top-left-radius: 16px;
  border-top-right-radius: 16px;
  padding: 12px 24px 26px 24px;
  z-index: 1;

  &__image {
    margin-right: 16px;
  }

  &__text {
    color: white;
    line-height: 24px;
    font-size: 18px;
    font-weight: bold;
  }
}
