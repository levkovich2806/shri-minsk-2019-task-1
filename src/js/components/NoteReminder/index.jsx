import React from "react";
import moment from "moment";

import styles from "./index.module.scss";

function getReadableData(date) {
  const normalDate = moment.unix(date);
  const time = moment.unix(date).format("HH:mm");
  const days = moment().diff(normalDate, "days");
  if (days > 0) {
    if (days === 1) {
      return `Завтра в ${time}`;
    }
    return `Осталось ${days} дня(дней)`;
  } else {
    return `Сегодня в ${time}`;
  }
}

function NoteReminder(props) {
  return (
    <div className={styles.reminder}>
      <div className={styles.reminder__image}>
        <img src="./images/reminder20.png" alt="Напоминание" />
      </div>
      <div className={styles.reminder__text}>
        {getReadableData(props.reminder)}
      </div>
    </div>
  );
}

export default NoteReminder;
