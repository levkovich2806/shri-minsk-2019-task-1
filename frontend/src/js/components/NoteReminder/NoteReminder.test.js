import React from "react";
import NoteReminder from "./index.jsx";
import renderer from "react-test-renderer";

describe("NoteReminder", function () {
  test("NoteReminder rendering", () => {

    const component = renderer.create(<NoteReminder reminder={1555527778000} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
