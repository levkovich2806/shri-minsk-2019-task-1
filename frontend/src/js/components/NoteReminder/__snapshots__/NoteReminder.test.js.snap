// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteReminder NoteReminder rendering 1`] = `
<div
  className="reminder"
>
  <div
    className="reminder__image"
  >
    <img
      alt="Напоминание"
      src="./images/reminder20.png"
    />
  </div>
  <div
    className="reminder__text"
  >
    Прошло 2 дня(дней)
  </div>
</div>
`;
